hummingbot/strategy/avellaneda_market_making/avellaneda_market_making.pyx

```diff
@@ -409,6 +424,84 @@ cdef class AvellanedaMarketMakingStrategy(StrategyBase):
+        self.get_config_map_indicators()
+
+    def get_config_map_execution_mode(self):
+        try:
+            execution_mode = self._config_map.execution_timeframe_mode.title
+            execution_timeframe = self._config_map.execution_timeframe_mode.Config.title
+            if execution_mode == FromDateToDateModel.Config.title:
+                start_time = self._config_map.execution_timeframe_mode.start_datetime
+                end_time = self._config_map.execution_timeframe_mode.end_datetime
+                execution_state = RunInTimeConditionalExecutionState(start_timestamp=start_time, end_timestamp=end_time)
+            elif execution_mode == DailyBetweenTimesModel.Config.title:
+                start_time = self._config_map.execution_timeframe_mode.start_time
+                end_time = self._config_map.execution_timeframe_mode.end_time
+                execution_state = RunInTimeConditionalExecutionState(start_timestamp=start_time, end_timestamp=end_time)
+            else:
+                start_time = None
+                end_time = None
+                execution_state = RunAlwaysExecutionState()
+
+            # Something has changed?
+            if self._execution_state is None or self._execution_state != execution_state:
+                self._execution_state = execution_state
+                self._execution_mode = execution_mode
+                self._execution_timeframe = execution_timeframe
+                self._start_time = start_time
+                self._end_time = end_time
+        except AttributeError:
+            self.logger().info("A parameter is missing in the execution timeframe mode configuration.")
+
+    def get_config_map_hanging_orders(self):
+        if self._config_map.hanging_orders_mode.title == TrackHangingOrdersModel.Config.title:
+            hanging_orders_enabled = True
+            hanging_orders_cancel_pct = self._config_map.hanging_orders_mode.hanging_orders_cancel_pct
+        else:
+            hanging_orders_enabled = False
+            hanging_orders_cancel_pct = Decimal("0")
+
+        if self._hanging_orders_enabled != hanging_orders_enabled:
+            # Hanging order tracker instance doesn't exist - create from scratch
+            self._hanging_orders_enabled = hanging_orders_enabled
+            self._hanging_orders_cancel_pct = hanging_orders_cancel_pct
+            self._hanging_orders_tracker = HangingOrdersTracker(self,
+                                                                hanging_orders_cancel_pct / Decimal('100'))
+            self._hanging_orders_tracker.register_events(self.active_markets)
+        elif self._hanging_orders_cancel_pct != hanging_orders_cancel_pct:
+            # Hanging order tracker instance existst - only update variable
+            self._hanging_orders_cancel_pct = hanging_orders_cancel_pct
+            self._hanging_orders_tracker.hanging_orders_cancel_pct = hanging_orders_cancel_pct / Decimal('100')
```

Comment:
Not really, the `_hanging_orders_cancel_pct ` is filled with the default value (10) after selecting the `track_hanging_orders` for `hanging_orders_mode`, and it's not being queried automatically right after that. Unlike with the execution mode. Therefore it can never throw an exception.

Reply:
Hmm I see. This is one more reason to implement this temporary "freezing" of the configs while the user is working with the `config_command` to avoid having the strategy operate momentarily with parameters the user potentially doesn't want it to operate with (i.e. the default values). But point taken that no try-catch needed here.